### PR TITLE
Restructuring work for BGP

### PR DESCRIPTION
**Pull request by roverflow**
Restructuring work for BGP

### CODE REVIEW

File: roles/deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+network.bgp.deploy
```
Comment (rohitthakur2590) [inline roles/deploy/README.md:1]: Can we add more details here

File: roles/detect/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+network.bgp.detect
```
Comment (rohitthakur2590) [inline roles/detect/README.md:1]: Can we add more details here

File: roles/gather/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+network.bgp.gather
```
Comment (rohitthakur2590) [inline roles/gather/README.md:1]: Can we add more details here

File: roles/persist/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+network.bgp.persist
```
Comment (rohitthakur2590) [inline roles/persist/README.md:1]: Can we add more details here

File: roles/remediate/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+network.bgp.remediate
```
Comment (rohitthakur2590) [inline roles/remediate/README.md:1]: Can we add more details here

File: roles/remediate/tasks/main.yaml [hunk 6]
```diff
@@ -1,11 +1,11 @@
 ---
 - name: Resolve target resources
-  ansible.builtin.include_tasks: includes/resources.yaml
+  ansible.builtin.include_tasks: resources.yaml
```
Comment (rohitthakur2590) [inline roles/remediate/tasks/main.yaml:3]: why do we still need this directory ?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: roles/configure/README.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+network.bgp.configure
+================
```
Comment (rohitthakur2590) [inline roles/configure/README.md:2]: This is where we need to provide documentation for description , examples and variables  for the role

File: roles/configure/defaults/main.yml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (rohitthakur2590) [inline roles/configure/defaults/main.yml:1]: we can get rid of directories which are empty , or has empty files

File: roles/configure/handlers/main.yml [hunk 9]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (rohitthakur2590) [inline roles/configure/handlers/main.yml:1]: If not needed lets remove handlers directory

File: roles/configure/tasks/main.yaml [hunk 10]
```diff
@@ -3,4 +3,4 @@
   ansible.builtin.include_role:
     name: network.base.resource_manager
   vars:
-    operation: 'configure'
+    operation: "configure"
```
Comment (rohitthakur2590) [inline roles/configure/tasks/main.yaml:6]: so in main.yaml can we include validation task and then may be including task which points to configure.yaml file with this task
for example : https://github.com/rohitthakur2590/network.backup/blob/main/roles/backup/tasks/main.yaml

File: roles/configure/vars/main.yml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+---
+# vars file for bgp_configure
```
Comment (rohitthakur2590) [inline roles/configure/vars/main.yml:2]: if not needed get rid of directory

File: roles/deploy/tasks/main.yaml [hunk 12]
```diff
@@ -1,10 +1,10 @@
 ---
 - name: Resolve target resources
-  ansible.builtin.include_tasks: includes/resources.yaml
```
Comment (rohitthakur2590) [inline roles/deploy/tasks/main.yaml:3]: same as for the configure role

File: roles/health_checks/README.md [hunk 13]
```diff
@@ -0,0 +1,4 @@
+network.bgp.health_checks
+================
```
Comment (rohitthakur2590) [inline roles/health_checks/README.md:2]: same comment as for configure role documentation

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]
I believe the var names we have used in the tasks , should be appended with role name.

File: roles/deploy/README.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+
+## Usage
+
+### Example 1: Deploy Configuration from Local Data Store
+Below example playbook demonstrates how to use the `bgp.deploy` role, where we will retrieve config from mentioned local data_store and deploy on to the network.
+
+```yaml
+- name: Deploy changes
+  hosts: rtr1
```
Comment (rohitthakur2590) [inline roles/deploy/README.md:26]: lets have hosts: all

File: roles/health_checks/tasks/main.yaml [hunk 15]
```diff
@@ -1,6 +1,6 @@
 ---
 - name: Include tasks
-  ansible.builtin.include_tasks: includes/{{ operation.name }}.yaml
+  ansible.builtin.include_tasks: "{{ operation.name }}.yaml"
```
Comment (rohitthakur2590) [inline roles/health_checks/tasks/main.yaml:3]: why we need operation ? is it relevant anymore ?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: roles/health_checks/tasks/main.yaml (hunk 15 above)
Comment (roverflow) [reply roles/health_checks/tasks/main.yaml]: Removed it and invoked the task directly

File: roles/health_checks/README.md (hunk 13 above)
Comment (rohitthakur2590) [reply roles/health_checks/README.md:2]: @roverflow  Can we have this Documentation in same format as we have other roles

**Review by rohitthakur2590 — APPROVED** [review]
Minor Comment , otherwise LGTM!

**Pull request merged**